npe2/manifest_cli.py

```diff
@@ -0,0 +1,45 @@
+from napari.plugins._plugin_manager import NapariPluginManager
+
+from npe2.manifest import PluginManifest
+
+
+def create_manifest(plugin_name: str, plugin_manager: None) -> str:
+    """function that takes plugin name and exports manifest"""
+
+    if plugin_manager is None:
+        plugin_manager = NapariPluginManager()
+
+    for current_name, plugin in sorted(
+        plugin_manager.plugins.items(), key=lambda x: x[0]
+    ):
+        if current_name == plugin_name:
+            name = plugin_name
+            publisher = plugin_manager.get_metadata(plugin, "author")
+            display_name = plugin_manager.get_standard_metadata(plugin_name)["package"]
+            description = plugin_manager.get_metadata(plugin, "summary")
+            version = plugin_manager.get_metadata(plugin, "version")
+
+            commands = []
+            for hook in plugin_manager.get_standard_metadata(plugin_name)["hooks"]:
+                package = plugin_manager.get_standard_metadata(plugin_name)["package"]
+                plugin_manager.get_standard_metadata(plugin_name)["hooks"]
+                hk = hook.split("napari_")[1]
+                id = package + "." + hk
+                python_name = plugin_manager.plugins[plugin_name].__name__ + ":" + hook
+                title = " ".join(hook.split("_"))
+                commands.append({"id": id, "python_name": python_name, "title": title})
+
+            contributions = {"commands": commands}
+
+            pm = PluginManifest(
+                name=name,
+                publisher=publisher,
+                display_name=display_name,
+                description=description,
+                version=version,
+                contributions=contributions,
+            )
+
+            return pm.yaml()
+
+    return None
```

Comment:
```suggestion
from npe2.manifest import PluginManifest


def create_manifest(plugin_name: str) -> str:
    """function that takes plugin name and exports manifest"""

    from napari.plugins import plugin_manager as plugin_manager

    plugin_manager.discover()

    if plugin_name not in plugin_manager.plugins:
        raise ValueError("Could not find plugin", plugin_name)

    module = plugin_manager.plugins[plugin_name]
    standard_meta = plugin_manager.get_standard_metadata(plugin_name)
    package = standard_meta["package"].replace("-", "_")

    commands = []
    for caller in plugin_manager._plugin2hookcallers[module]:
        for impl in caller.get_hookimpls():
            if impl.plugin_name != plugin_name:
                continue
            name = impl.specname.replace("napari_", "")
            id = f"{package}.{name}"
            python_name = f"{impl.function.__module__}:{impl.function.__qualname__}"
            title = " ".join(name.split("_")).title()
            commands.append({"id": id, "python_name": python_name, "title": title})

    pm = PluginManifest(
        name=package,
        publisher=standard_meta["author"],
        description=standard_meta["summary"],
        version=standard_meta["version"],
        contributions={"commands": commands},
    )
    return pm.yaml()
```

i was just playing with this locally, sorry for the large suggestion.  here are the bullet points:

- don't import napari until inside of the function
- let's not accept a plugin manager, just always use the napari.plugins.plugin_manager
- we should call `discover()` manually to make sure that everything is found before we start
- we should raise an exception if something isn't found (rather than returning None)
- the manifest name should be the package name, _not_ the current plugin name in napari (confusing)
- I'm iterating through `plugin_manager._plugin2hookcallers[module]` rather than `plugin_manager.get_standard_metadata(plugin_name)["hooks"]` ... the main reason is that it's conceivable that a plugin has used a different name than the hookspec (i.e. you don't _have_ to name your function `napari_get_reader` ... you can actually name it something else...), so we really need to grab the exact handle to the function, which only the hookCaller can give us.

everything else looks good!  

you should start to install various plugins and then do things like 

```py
from npe2.manifest_cli import create_manifest
from napari.plugins import plugin_manager

plugin_manager.discover()

for p in plugin_manager.plugins:
    try:
        mf = create_manifest(p)
        print(p, 'worked!')
    except Exception as e:
        print(p, "failed", e)
```